src/main/java/edu/ucsb/mapache/controllers/SlackSlashCommandController.java

```diff
@@ -0,0 +1,259 @@
+                params.getCommand(), params.getTextParts()[0]));
+        richMessage.setResponseType("ephemeral");
+        return richMessage.encodedMessage(); // don't forget to send the encoded message to Slack
+    }
+
+    public RichMessage emptyCommand(SlackSlashCommandParams params) {
+        RichMessage richMessage = new RichMessage(
+                String.format("The %s bot knows these commands", params.getCommand()));
+
+        richMessage.setResponseType("ephemeral");
+
+        int numAttachments = 3;
+        Attachment[] attachments = new Attachment[numAttachments];
+        for (int i = 0; i < numAttachments; i++)
+            attachments[i] = new Attachment();
+        attachments[0].setText(String.format("%s status", params.getCommand()));
+        attachments[1].setText(String.format("%s time", params.getCommand()));
+        attachments[2].setText(String.format("%s debug", params.getCommand()));
+        richMessage.setAttachments(attachments);
+
+        return richMessage.encodedMessage(); // don't forget to send the encoded message to Slack
+    }
+
+    public RichMessage statusCommand(SlackSlashCommandParams params) {
+        RichMessage richMessage = new RichMessage(
+                String.format("From: %s Status is ok!", params.getCommand(), params.getTextParts()[0]));
+        richMessage.setResponseType("ephemeral");
+
+        
+        return richMessage.encodedMessage(); // don't forget to send the encoded message to Slack
+    }
+
+    public RichMessage googleSearch(SlackSlashCommandParams params) { // google search
+        // richMessage.setResponseType("ephemeral");
+        String[] textParts = params.getTextParts();
+        String finResults = "";
+        try { 
+            String baseURL = "https://www.google.com/search?q=";
+            
+            // make new baseURL using getTextParts as input;
+            // assuming input is formatted as /mapache googleSearch "input words" // textparts is a java array 
+            String attachments = "";
+            for (int i = 2; i < textParts.length; i++) {
+                if (i != 2) {attachments += "+";}
+                attachments += textParts[i];
+            }
+            baseURL = baseURL + attachments;
+            Document searchPage = Jsoup.connect(baseURL).get();
+
+            Elements links = searchPage.select("div.J9WfR eqAnXb");
+            links = searchPage.select("a[href]");
+            //
+
+            ArrayList<String> linkArrays = new ArrayList<String>();
+
+            for (Element l : links) {
+				String temp = l.attr("abs:href");
+				if (temp.contains("https://www.google.com") == false &&
+						temp.contains("https://www.youtube.com") == false &&
+						temp.contains("https://support.google.com") == false &&
+						temp.contains("https://maps.google.com") == false &&
+						temp.contains("https://policies.google.com") == false) {
```

Comment:
Comparing booleans is redundant, could we use `!temp.contains` instead?

Reply:
Going to leave it as is due that in my personal opinion, it seems to be more readable in this way.